iceoryx_posh/include/iceoryx_posh/internal/popo/modern_api/base_subscriber.inl

```diff
@@ -104,46 +107,69 @@ inline cxx::expected<cxx::optional<Sample<const T>>, ChunkReceiveError> BaseSubs
 {
-    m_port.setConditionVariable(conditionVariableDataPtr);
+    if (m_trigger.getUniqueId() == uniqueTriggerId)
+    {
+        m_port.unsetConditionVariable();
+        m_trigger.invalidate();
+    }
 }
 
-template <typename T, typename port_t>
-inline void BaseSubscriber<T, port_t>::unsetConditionVariable() noexcept
+// ============================== Sample Deleter ============================== //
+
+template <typename T, typename Subscriber, typename port_t>
+inline BaseSubscriber<T, Subscriber, port_t>::SubscriberSampleDeleter::SubscriberSampleDeleter(port_t& port)
+    : m_port(std::ref(port))
 {
-    m_port.unsetConditionVariable();
 }
 
-template <typename T, typename port_t>
-inline bool BaseSubscriber<T, port_t>::hasTriggered() const noexcept
+template <typename T, typename Subscriber, typename port_t>
+inline void BaseSubscriber<T, Subscriber, port_t>::SubscriberSampleDeleter::operator()(T* const ptr) const
 {
-    return m_port.hasNewChunks();
+    auto header = mepoo::convertPayloadPointerToChunkHeader(reinterpret_cast<void*>(ptr));
+    m_port.get().releaseChunk(header);
 }
 
-// ============================== Sample Deleter ============================== //
-
-template <typename T, typename port_t>
-inline BaseSubscriber<T, port_t>::SubscriberSampleDeleter::SubscriberSampleDeleter(port_t& port)
-    : m_port(std::ref(port))
+template <typename T, typename Subscriber, typename port_t>
+inline cxx::expected<WaitSetError>
+BaseSubscriber<T, Subscriber, port_t>::attachTo(WaitSet& waitset,
+                                                [[gnu::unused]] const SubscriberEvent subscriberEvent,
+                                                const uint64_t triggerId,
+                                                const Trigger::Callback<Subscriber> callback) noexcept
 {
+    using SelfType = BaseSubscriber<T, Subscriber, port_t>;
+    Subscriber* self = reinterpret_cast<Subscriber*>(this);
+
+    return waitset
+        .acquireTrigger(self,
+                        {this, &BaseSubscriber<T, Subscriber, port_t>::hasNewSamples},
+                        {this, &SelfType::invalidateTrigger},
+                        triggerId,
+                        callback)
+        .and_then([this](TriggerHandle& trigger) {
+            m_trigger = std::move(trigger);
+            m_port.setConditionVariable(m_trigger.getConditionVariableData());
+        });
 }
 
-template <typename T, typename port_t>
-inline void BaseSubscriber<T, port_t>::SubscriberSampleDeleter::operator()(T* const ptr) const
+template <typename T, typename Subscriber, typename port_t>
+inline void BaseSubscriber<T, Subscriber, port_t>::detachEvent(const SubscriberEvent subscriberEvent) noexcept
 {
-    auto header = mepoo::convertPayloadPointerToChunkHeader(reinterpret_cast<void*>(ptr));
-    m_port.get().releaseChunk(header);
+    static_cast<void>(subscriberEvent);
```

Comment:
Ok, for now the event is not relevant. I guess because there is just one. What would happen if there would be more and it was relevant? (still trying to figure out the design intention here).

Reply:
It is described in detail in the `ice_waitset_trigger.cpp` example. The idea is that you would have then multiple triggerHandle members and would assign the acquired triggerHandle to the appropriate handle.